tracer/build/_build/CodeGenerators/CallTargetsGenerator.cs

```diff
@@ -0,0 +1,743 @@
+                                        .ThenBy(static x => x.Key.TargetTypeName)
+                                        .ThenBy(static x => x.Key.TargetMethodName))
+            {
+                sb.AppendLine(GetCallTarget(definition.Key, signatures[GetSignature(definition.Key)], definition.Value, x++));
+            }
+
+            sb.AppendLine("""
+                };
+                }
+                """);
+
+
+            if (!Directory.Exists(outputPath)) { Directory.CreateDirectory(outputPath); }
+            var fileName = outputPath / "generated_calltargets.g.h";
+            File.WriteAllText(fileName, sb.ToString());
+
+            Serilog.Log.Information("CallTarget definitions File saved: {File}", fileName);
+
+            static string GetSignature(CallTargetDefinitionSource definition)
+            {
+                string sig = $"(WCHAR*)WStr(\"{definition.TargetReturnType}\"),";
+                foreach (var arg in definition.TargetParameterTypes)
+                {
+                    sig += $"(WCHAR*)WStr(\"{arg}\"),";
+                }
+
+                return sig;
+            }
+
+            static string GetSignatureName(int index)
+            {
+                return $"g_callTargets_Sig_{index:000}";
+            }
+
+            static string GetSignatureField(string signatureName, string signature)
+            {
+                return $"WCHAR* {signatureName}[]={{{signature}}};";
+            }
+
+            static string GetCallTarget(CallTargetDefinitionSource definition, string signature, TargetFrameworks tfms, int index)
+            {
+                var min = definition.MinimumVersion;
+                var max = definition.MaximumVersion;
+
+                var typ = $"(WCHAR*)WStr(\"{definition.AssemblyName}\"),(WCHAR*)WStr(\"{definition.TargetTypeName}\"),(WCHAR*)WStr(\"{definition.TargetMethodName}\"),";
+                var sig = $"{signature},{definition.TargetParameterTypes.Count + 1},";
+                var ver = $"{min.Major},{min.Minor},{min.Patch},{max.Major},{max.Minor},{max.Patch},";
+                var asy = $"assemblyName,(WCHAR*)WStr(\"{definition.InstrumentationTypeName}\"),{GetCallTargetKind(definition.IntegrationKind)},{(int)definition.InstrumentationCategory}";
+                var tfm = $",{(uint)tfms}";
+                return $"{{{typ}{sig}{ver}{asy}{tfm}}},";
+            }
+
+            static string GetCallTargetKind(int kind)
+            {
+                /*
+                 enum class CallTargetKind : UINT8
+                {
+                    Default = 0,
+                    Derived = 1,
+                    Interface = 2
+                };
+                 */
+                return kind switch
+                {
+                    0 => "CallTargetKind::Default",
+                    1 => "CallTargetKind::Derived",
+                    2 => "CallTargetKind::Interface",
+                    _ => "ERROR"
```

Comment:
We can (should) just throw here I think? Or we could throw where this is called, so we have more context?

```suggestion
                    _ => throw new InvalidOperationException($"Invalid call target kind: {kind}" );
```